packages/ai/src/ai/common/validation.py

```
@@ -0,0 +1,147 @@
+    if not prompt.strip():
+        raise ValueError('Prompt is empty after sanitization.')
+
+    # Check token count - warn but don't block (ChatBase.chat_string already
+    # has a softer check; this catches the truly egregious cases early)
+    try:
+        token_count = token_counter(prompt)
+        if token_count > max_tokens:
+            debug(f'Warning: Prompt ({token_count} tokens) exceeds model context window ({max_tokens} tokens). The request will likely be rejected by the provider.')
+    except Exception:
+        # Token counting failures should not block the request
+        pass
```

Ruff (0.15.7)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)